lattices/src/algebra.rs

```diff
@@ -348,6 +348,59 @@ pub fn idempotency<S: Debug + PartialEq + Clone, const N: usize>(
     Ok(())
 }
 
+/// Defines the linearity property
+/// q is linear with respect to some group operation + if q(a+b) = q(a) + q(b)
+/// This is the same as q being a group homomorphism
+// As defined in the paper "DBSP: Automatic Incremental View Maintenance for Rich Query Languages"
+pub fn linearity<S: Debug + PartialEq + Clone, R: Debug + PartialEq + Clone, const N: usize>(
+    items: &[S; N],
```

Comment:
```rust
pub fn linearity<S: Debug + PartialEq + Clone, R: Debug + PartialEq + Clone>(
    items: &[S],
```
is simpler and more general - `&[Item; N]` is a reference to a length `N` array, `&[Item]` is a reference to any length array (slice)

Reply:
fixed